### PR TITLE
Prevent double-observers for the idleTimerDisabled value change

### LINKED ISSUE #107: EXC_BAD_ACCESS when triggering wakelock on iOS
We're currently investigating a problem with this plugin when used on iOS. 
The symptom is a full app crash with EXC_BAD_ACCESS, in the iOS side's `observeValueForKeyPath` method:

Screenshot 2021-04-15 at 21 56 25

The stacktrace points to a recursion:
Screenshot 2021-04-15 at 21 56 58

Our App actually runs two isolates, which in turn means **two** instances of the wakelock plugin are active and interfering with each other.

Screenshot 2021-04-15 at 21 58 25

This could be resolved by setting up single observer instead of one per plugin instance.

### PR DESCRIPTION
**Pull request by ened**
Add a additional class to wrap observers for the sharedApplication.idleTimerDisabled

## Description

Closes #107 by ensuring only a single observer for the idle timer runs, regardless of how many wakelock plugin instances exist in the system.

### CODE REVIEW

File: wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -2,6 +2,6 @@
 <Workspace
    version = "1.0">
    <FileRef
-      location = "group:Runner.xcodeproj">
+      location = "self:">
```
Comment (creativecreatorormaybenot) [inline wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: I am not familiar with this change. Can you explain why it is included?

**Review by creativecreatorormaybenot — COMMENTED** [review]
Thanks for the contribution! LGTM

Can you address the comment and add a CHANGELOG entry + version bump in `pubspec.yaml`? :)  
When that is done, I will approve and once CI checks succeed, I think we can merge it.

**Comment by creativecreatorormaybenot** [comment]
All checks are successful - beautiful :) We do run integration tests on iOS in this repo, so the change should be covered.

File: wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 1 above)
Comment (ened) [reply wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: Xcode 12.x did that change automatically - I had this in a few projects recently, so should be fine.

Comment (creativecreatorormaybenot) [reply wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: @ened I figured that, makes sense. Do you want to add the changelog entry or should I do it?

Comment (ened) [reply wakelock/example/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: On it.

**Comment by creativecreatorormaybenot** [comment]
@ened Thanks so much for the contribution!

**Comment by creativecreatorormaybenot** [comment]
@ened your commits are not GPG signed. it is fine, I will merge anyway (of course). Just thought I would mention it in case you are interested 👍

**Comment by ened** [comment]
@creativecreatorormaybenot ok thank you. Could you please publish to pub.dev ?

**Comment by creativecreatorormaybenot** [comment]
I am so sorry @ened - the CD takes care of publishing. But since I changed the branch name from `master` to `main`, it did not run and after fixing it, it did not pick up the change that was done.

I published it now 👍

**Pull request merged**